From 80bd6dd83f8b63e1b47defdc4b52d4e817724f43 Mon Sep 17 00:00:00 2001
Message-Id: <80bd6dd83f8b63e1b47defdc4b52d4e817724f43.1427265330.git.chang-joon.lee@intel.com>
In-Reply-To: <b4c66aa6a04d2be0fca9fa4448c7188a581a57a5.1427265330.git.chang-joon.lee@intel.com>
References: <b4c66aa6a04d2be0fca9fa4448c7188a581a57a5.1427265330.git.chang-joon.lee@intel.com>
From: Gajanan Bhat <gajanan.bhat@intel.com>
Date: Tue, 10 Mar 2015 16:44:57 +0530
Subject: [PATCH 07/22] REVERTME [VPG]: drm/i915: Remove DSPCNTR programming
 from crtc enable sequence

Dont write to DSPCNTR register in crtc enable function before enabling pipes.
This register is a trigger register. Triggering this register before enabling
pipe can lead to pipe hang.

REVERTME: This patch is marked as revertme since this change is already part
of upstream. This change can be discarded after next forklift.
Since the upstream patch mentioned below includes lot more
changes than this we not forklifting the patch right away.
Upstream-Repo: git://anongit.freedesktop.org/drm-intel
Commit-ID: f45651bae2ee73ae551699d481f76aa6ad92138f
Commit-Message: "drm/i915: Eliminate rmw from .update_primary_plane()".

Issue: GMINL-6910
Change-Id: Iabc881f9a2dfbdcca1a19c53dcf34953b16cf760
Signed-off-by: Gajanan Bhat <gajanan.bhat@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c |    7 -------
 1 file changed, 7 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 7df27a8..b4a0fdd 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -5348,7 +5348,6 @@ static void valleyview_crtc_enable(struct drm_crtc *crtc)
 	int pipe = intel_crtc->pipe;
 	int plane = intel_crtc->plane;
 	bool is_dsi;
-	u32 dspcntr;
 
 	WARN_ON(!crtc->enabled);
 
@@ -5360,9 +5359,6 @@ static void valleyview_crtc_enable(struct drm_crtc *crtc)
 	if (!is_dsi && !IS_CHERRYVIEW(dev))
 		vlv_prepare_pll(intel_crtc);
 
-	/* Set up the display plane register */
-	dspcntr = DISPPLANE_GAMMA_ENABLE;
-
 	if (intel_crtc->config.has_dp_encoder ||
 		intel_pipe_has_type(&intel_crtc->base, INTEL_OUTPUT_EDP))
 		intel_dp_set_m_n(intel_crtc);
@@ -5378,9 +5374,6 @@ static void valleyview_crtc_enable(struct drm_crtc *crtc)
 
 	i9xx_set_pipeconf(intel_crtc);
 
-	I915_WRITE(DSPCNTR(plane), dspcntr);
-	POSTING_READ(DSPCNTR(plane));
-
 	dev_priv->display.update_primary_plane(crtc, crtc->primary->fb,
 					       crtc->x, crtc->y);
 
-- 
1.7.9.5

